agent/consul/server.go

```diff
@@ -1655,6 +1655,8 @@ func (s *Server) trackLeaderChanges() {
 			}
 
 			s.grpcLeaderForwarder.UpdateLeaderAddr(s.config.Datacenter, string(leaderObs.LeaderAddr))
+			// TODO(peering): might need to send DC or use DC on the peering backend in the future
```

Comment:
What do you mean by this? DC names are largely not relevant in the context of peering

Reply:
I was chatting with RB and maybe mesh gateways may need to know this so that's why I dropped this comment; I'm also open to removing it if it's causing confusion

Reply:
I lean toward removing it, if it's needed we likely don't need this TODO as a reminder